keras/layers/core.py

```diff
@@ -319,17 +319,16 @@ class Masking(MaskedLayer):
     def __init__(self, mask_value=0., **kwargs):
         super(Masking, self).__init__(**kwargs)
         self.mask_value = mask_value
-        self.input = K.placeholder(ndim=3)
+        if (not hasattr(self, 'input')):
```

Comment:
It is set by the parent class when you specify the `input_shape`, so us overriding it afterwards was causing issues in some tests.